enos/modules/vault-verify-write-data/templates/smoke-enable-secrets-kv.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+set -e
+
+function retry {
+  local retries=$1
+  shift
+  local count=0
+
+  until "$@"; do
+    exit=$?
+    wait=$((2 ** count))
+    count=$((count + 1))
+    if [ "$count" -lt "$retries" ]; then
+      sleep "$wait"
+    else
+      return "$exit"
+    fi
+  done
+
+  return 0
+}
+
+function fail {
+	echo "$1" 1>&2
+	exit 1
+}
+
+binpath=${vault_install_dir}/vault
+instanceid=${instance_id}
+
+# Only enable the secrets engine on the first node
+if test $instanceid != '0'; then
```

Comment:
Perhaps you can remove this and make the `remote_exec` that calls this script only execute on the first instance. It means you can also remove the `instanceid` variable. The more I think about it the more I really feel this doesn't belong here. Reason being, if I was to use this script via the module and it doesn't blowup since it returns `exit 0` and doesn't enable the secrets kv, I would be scratching my head. Also, even it it doesn't take that much time to run a `remote_exec` task, it's unnecessary and could fail.

Reply:
Good call, I took this module directly from the old smoke tests and hadn't noticed this. Change made!